### PR TITLE
feat(forwarder): add `debug_endpoint` parameter

### PR DESCRIPTION
**Pull request by jta**

### CODE REVIEW

**Review by obs-gh-alexlew — APPROVED** [review]

File: modules/forwarder/variables.tf [hunk 1]
```diff
@@ -155,3 +155,10 @@ variable "queue_maximum_batching_window_in_seconds" {
   nullable    = false
   default     = 1
 }
+
+variable "debug_endpoint" {
+  description = "Endpoint to send debugging telemetry to. Sets the OTEL_EXPORTER_OTL_ENDPOINT environment variable for the lambda function."
```
Comment (obs-gh-abhishekrao) [inline modules/forwarder/variables.tf:160]: *OTLP

File: modules/forwarder/lambda.tf [hunk 2]
```diff
@@ -11,10 +11,11 @@ resource "aws_lambda_function" "this" {
...
     variables = merge({
-      DESTINATION_URI        = "s3://${var.destination.bucket}/${var.destination.prefix}"
-      MAX_FILE_SIZE          = var.max_file_size
-      CONTENT_TYPE_OVERRIDES = join(",", [for o in var.content_type_overrides : "${o["pattern"]}=${o["content_type"]}"])
-      SOURCE_BUCKET_NAMES    = join(",", var.source_bucket_names)
+      DESTINATION_URI             = "s3://${var.destination.bucket}/${var.destination.prefix}"
+      MAX_FILE_SIZE               = var.max_file_size
+      CONTENT_TYPE_OVERRIDES      = join(",", [for o in var.content_type_overrides : "${o["pattern"]}=${o["content_type"]}"])
+      SOURCE_BUCKET_NAMES         = join(",", var.source_bucket_names)
+      OTEL_EXPORTER_OTLP_ENDPOINT = var.debug_endpoint
```
Comment (obs-gh-abhishekrao) [inline modules/forwarder/lambda.tf:18]: I don't see an option to specify any token or credentials for this endpoint. Is this an Observe endpoint or any generic OTLP endpoint?

Comment (jta) [reply modules/forwarder/lambda.tf:18]: It can be any OTLP compatible endpoint.

File: modules/forwarder/variables.tf (hunk 1 above)
Comment (jta) [reply modules/forwarder/variables.tf:160]: Good catch

File: modules/forwarder/lambda.tf (hunk 2 above)
Comment (obs-gh-abhishekrao) [reply modules/forwarder/lambda.tf:18]: Fair, but assuming our customers / prospects are primary consumers of this, we will likely get asked "where do I put my datastream token for this endpoint?". Do we have an answer for that?

Comment (jta) [reply modules/forwarder/lambda.tf:18]: You put it in via basic auth. This can get documented separately.

**Pull request merged**